api/app/Extensions/Revisionable/ChangeloggableTrait.php

```diff
@@ -0,0 +1,282 @@
+<?php
```

Comment:
@artstorm yes definitely, but I think with the below suggestions, it is possible to make the trait completely decoupled - all that would be needed is to add the `RevisionableTrait`